File: src/005/p005.scala
object Main {
  def gcd(a: Int, b: Int): Int = {
    if (b == 0) return a.abs
    gcd(b, a % b)
  }

  def lcm(a: Int, b: Int): Int = {
    (a * b).abs / gcd(a, b)
  }

  def solve(): Int = {
    (1 to 19).reduce((acc, x) => lcm(acc, x))
  }

  def main(args: Array[String]): Unit = {
    val result = solve()
    println(result)
  }
}

File: src/005/p005.exs
defmodule Problem005 do
  defp gcd(x, 0), do: x
  defp gcd(x, y), do: gcd(y, rem(x, y))
  defp lcm(x, y), do: x * y / gcd(x, y)

  def solve do
    1..20
    |> Enum.reduce(fn(x, acc) -> round(lcm(x, acc)) end)
  end
end

IO.puts Problem005.solve
